DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -389,41 +383,81 @@ Function Upgrade-Package {
     if ($cParams) {
         $chocoupgradeparams += " $cParams"
     }
-    $cmd = "choco upgrade $pName $chocoupgradeparams"
+    $cmd = "upgrade $pName $chocoupgradeparams"
     Write-Verbose -Message "Upgrade command: '$cmd'"
 
     if (-not (IsPackageInstalled -pName $pName))
     {
         throw "$pName is not installed, you cannot upgrade"
     }
 
-    $packageUpgradeOuput = Invoke-Expression -Command $cmd
+    $packageUpgradeOuput = Invoke-Chocolatey $cmd
     $packageUpgradeOuput | ForEach-Object { Write-Verbose -Message $_ }
+}
 
-    # Clear Package Cache
-    Get-ChocoInstalledPackage 'Purge'
+function Get-ChocoInstalledPackage {
+    Return (choco list -lo -r | ConvertFrom-Csv -Header 'Name', 'Version' -Delimiter "|")
 }

```

Comment:
What's your thinking behind running this every time rather than caching it for 60 seconds?
The Return isn't needed.

Reply:
In some instances of DSC configuratioin 60 seconds is to long if a small package is installed, so the test will fail I will remove the Return

Reply:
See previous comment around this.